tsdb/index/index.go

```diff
@@ -796,13 +970,17 @@ func (w *Writer) Close() error {
 			return err
 		}
 	}
-	if err := w.fbuf.Flush(); err != nil {
-		return err
+	if w.fP != nil {
+		if err := w.fP.close(); err != nil {
```

Comment:
Not remove? and below?

Reply:
The only time it could be removed here is if something went wrong, in which case leaving it behind for debugging with the incomplete main index file seems better than adding more complex logic to remove it.